### PR TITLE
Extract `buildSchemaProperties` method in Delta Lake

### LINKED ISSUE #12030: DeltaLake extract buildSchemaProperties from CheckpointWriter
buildSchemaProperties is implemented both in [CheckpointWriter](https://github.com/trinodb/trino/blob/2160c4b62c4cedf8bad876aba862e8847902979e/plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/checkpoint/CheckpointWriter.java#L147) and in [DeltaLakePageSink](https://github.com/trinodb/trino/blob/2160c4b62c4cedf8bad876aba862e8847902979e/plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakePageSink.java#L503) 
It could be extracted to utility class

### PR DESCRIPTION
**Pull request by ebyhr**
## Description

Extract `buildSchemaProperties` method in Delta Lake
Fixes #12030

## Documentation

(x) No documentation is needed.

## Release notes

(x) No release notes entries required.

### CODE REVIEW

**Comment by ebyhr** [comment]
CI failed 
```
Error:  Tests run: 21, Failures: 1, Errors: 0, Skipped: 20, Time elapsed: 11.824 s <<< FAILURE! - in io.trino.faulttolerant.delta.TestDeltaTaskFailureRecoveryTest
Error:  io.trino.faulttolerant.delta.TestDeltaTaskFailureRecoveryTest.init  Time elapsed: 11.656 s  <<< FAILURE!
com.github.dockerjava.api.exception.InternalServerErrorException: 
Status 500: {"message":"Head \"[https://registry-1.docker.io/v2/testcontainers/ryuk/manifests/0.3.3\](https://registry-1.docker.io/v2/testcontainers/ryuk/manifests/0.3.3/)": Get \"https://auth.docker.io/token?account=githubactions&scope=repository%3Atestcontainers%2Fryuk%3Apull&service=registry.docker.io\": EOF"}
	at org.testcontainers.shaded.com.github.dockerjava.core.DefaultInvocationBuilder.execute(DefaultInvocationBuilder.java:247)
	at org.testcontainers.shaded.com.github.dockerjava.core.DefaultInvocationBuilder.lambda$executeAndStream$1(DefaultInvocationBuilder.java:269)
	at java.base/java.lang.Thread.run(Thread.java:829)
```

File: plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeTypeUtil.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+import java.util.List;
+import java.util.Properties;
+
+import static java.util.stream.Collectors.joining;
+
+public final class DeltaLakeTypeUtil
+{
+    private DeltaLakeTypeUtil() {}
+
+    public static Properties buildSchemaProperties(List<String> columnNames, List<Type> columnTypes)
```
Comment (findinpath) [inline plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeTypeUtil.java:30]: Note that there exists already a utility class called `DeltaLakeSchemaProperties`.

There exists in `trino-hive` also a similar utility method called `io.trino.plugin.hive.metastore.MetastoreUtil#getHiveSchema(io.trino.plugin.hive.metastore.Table)`


What about using similarly a name like `buildHiveSchema` ?

File: plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeTypeUtil.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+import io.trino.plugin.hive.HiveTypeName;
+import io.trino.spi.type.Type;
+import org.apache.hadoop.hive.ql.io.IOConstants;
+
+import java.util.List;
+import java.util.Properties;
+
+import static java.util.stream.Collectors.joining;
+
+public final class DeltaLakeTypeUtil
```
[the remark this answers is not in the document]
Comment (findinpath) [reply plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeTypeUtil.java:26]: Would the name `DeltaLakeHiveSchema` fit better the purpose of this class?

File: plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeTypeUtil.java (hunk 1 above)
Comment (ebyhr) [reply plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeTypeUtil.java]: Thanks, I was missing the class since it was not final.

**Pull request merged**